### DISCUSSION TITLE
Invalidate/Expire data manually

### DISCUSSION THREAD

**Discussion by th3nu11** [ZiggyCreatures/FusionCache#138] [Q&A] [answered]
Hi,

I have to manually invalidate the cache and now I'm using the Remove method. Since I'm using the FailSafe approach, I'd rather just invalidate/expire the data than remove it (this way the data will be loaded on the first request and if there is an error the old data will be returned).
Is there any way to do this?

I've searched the documentation (and code) but haven't found anything like this. I think the method I need is something similar to the MemoryCacheAccessor.EvictEntry method, but I'm not sure.

Thank you!

**Comment by jodydonetti** [comment]
HI @th3nu11 and thanks for using FusionCache!

A method to expire a cache entry without removing it was on my radar for some time: as you correctly pointed out it does already exist in the `MemoryCacheAccessor` internal class, but I decided back then not to expose it because I wanted to collect some more usage informations before doing it.

Maybe now is the time to do it.

Let me think a little bit about it and will come back to you with news soon.

**Reply by jodydonetti** [reply]
Hi @th3nu11 , I just dropped a new [issue](https://github.com/ZiggyCreatures/FusionCache/issues/146) for a new `Expire()` method.
Please let me know what you think, thanks!

**Comment by jodydonetti** [comment, marked as answer]
Hi @th3nu11 , [v0.22.0](https://github.com/ZiggyCreatures/FusionCache/releases/tag/v0.22.0) has been released and `Expire()` / `ExpireAsync()` is now officially supported 🎉
